java-generator/it/src/test/java/io/fabric8/java/generator/gradle/plugin/SimpleIT.java

```diff
@@ -0,0 +1,119 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.fabric8.java.generator.gradle.plugin;
+
+import org.apache.commons.io.FileUtils;
+import org.gradle.testkit.runner.BuildResult;
+import org.gradle.testkit.runner.UnexpectedBuildFailure;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.*;
+
+class SimpleIT {
+
+  @RegisterExtension
+  private final ITGradleRunnerExtension gradleRunner = new ITGradleRunnerExtension();
+
+  static final List<File> TESTED_CRDS = Stream.of(
```

Comment:
In other tests we are using the `junit` `TempDir`, e.g.:
https://github.com/fabric8io/kubernetes-client/blob/fe579f0eab6ece572cee8a184f0614864a8c95a6/java-generator/core/src/test/java/io/fabric8/java/generator/CompilationTest.java#L45-L46

would it be possible to apply the same pattern?

Reply:
applies at line 45, sorry for the confusion.

Reply:
I am wondering whether this is the right use case for `@TempDir`. Maybe I am missing some pieces but according to the docs its lifecycle would be managed by JUnit. Which clashes with the fact that those directories are instead managed by the Gradle build. `resources` is an exception, but I'll use it to store the CRD locally.

Reply:
Oh, if those are managed by Gradle it's fine, missed it 👍